multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -202,6 +202,7 @@ def __init__(
         enable_progress_bar: Optional[bool] = None,
         init_scratch: Optional[bool] = False,
         sample_data_path: Optional[str] = None,
+        clean_old_ckpts: Optional[bool] = True,
```

Comment:
What's the case of not cleaning the old checkpoints?

Reply:
While pretraining on a very large dataset (e.g. objects365), validation with original split may crash the process. So we may want to save the checkpoints and evaluate them on valid dataset after the training.

Reply:
Is the crash some kind of bug?

Reply:
Not really, there may be unavoidable OOM errors on RAM or GRAM.

Reply:
If OOM errors happen, do we need to turn the validation off in training?

Reply:
If the errors are random, we can resume training and still do validation.

Reply:
It's very possible to have OOM error if the validation set is very large. We can sample a small portion of validation data to give us some rough information about the training progress. But we don't delete the checkpoints and will do the full evaluation on validation set after training.

Reply:
Nick bring some concerns today tho that we may still need early stopping, and Alex also mentioned that we might want more advanced sampling options strategy than uniform. We could work on this in next PR.

Reply:
Validation generally uses less memory than training. What's the cause of OOM?